driver/src/main/java/org/neo4j/driver/internal/summary/InternalNotification.java

```diff
@@ -16,86 +16,86 @@
  */
 package org.neo4j.driver.internal.summary;
 
-import static org.neo4j.driver.internal.value.NullValue.NULL;
-
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
-import java.util.function.Function;
 import org.neo4j.driver.NotificationCategory;
+import org.neo4j.driver.NotificationClassification;
 import org.neo4j.driver.NotificationSeverity;
 import org.neo4j.driver.Value;
-import org.neo4j.driver.internal.InternalNotificationCategory;
-import org.neo4j.driver.internal.InternalNotificationSeverity;
 import org.neo4j.driver.summary.InputPosition;
 import org.neo4j.driver.summary.Notification;
 
-public class InternalNotification implements Notification {
-    public static final Function<Value, Notification> VALUE_TO_NOTIFICATION = value -> {
-        var code = value.get("code").asString();
-        var title = value.get("title").asString();
-        var description = value.get("description").asString();
-        var rawSeverityLevel =
-                value.containsKey("severity") ? value.get("severity").asString() : null;
-        var severityLevel =
-                InternalNotificationSeverity.valueOf(rawSeverityLevel).orElse(null);
-        var rawCategory = value.containsKey("category") ? value.get("category").asString() : null;
-        var category = InternalNotificationCategory.valueOf(rawCategory).orElse(null);
-
-        var posValue = value.get("position");
-        InputPosition position = null;
-        if (posValue != NULL) {
-            position = new InternalInputPosition(
-                    posValue.get("offset").asInt(),
-                    posValue.get("line").asInt(),
-                    posValue.get("column").asInt());
-        }
-
-        return new InternalNotification(
-                code, title, description, severityLevel, rawSeverityLevel, category, rawCategory, position);
-    };
+public class InternalNotification extends InternalGqlStatusObject implements Notification {
+    @SuppressWarnings("deprecation")
+    public static Optional<NotificationCategory> valueOf(String value) {
+        return Arrays.stream(NotificationClassification.values())
+                .filter(type -> type.toString().equals(value))
+                .findFirst()
+                .map(type -> switch (type) {
+                    case HINT -> NotificationCategory.HINT;
+                    case UNRECOGNIZED -> NotificationCategory.UNRECOGNIZED;
+                    case UNSUPPORTED -> NotificationCategory.UNSUPPORTED;
+                    case PERFORMANCE -> NotificationCategory.PERFORMANCE;
+                    case DEPRECATION -> NotificationCategory.DEPRECATION;
+                    case SECURITY -> NotificationCategory.SECURITY;
+                    case TOPOLOGY -> NotificationCategory.TOPOLOGY;
+                    case GENERIC -> NotificationCategory.GENERIC;
+                });
```

Comment:
How do you handle unknown here?

Reply:
An empty `Optional` is returned in such case.

Reply:
This is the way you already did with the legacy Notification, so this is fine.